### PR TITLE
feat(components/lookup): add search test harness

### PR DESCRIPTION
**Pull request by Blackbaud-SandhyaRajasabeson**
[AB#2195526](https://dev.azure.com/blackbaud/Products/_boards/board/t/SKY%20UX%20Program/Stories/?workitem=2195526)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2798/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_ 

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2798/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2798/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2798/playground/)

File: libs/components/lookup/testing/src/search/search-harness.ts
Comment (Blackbaud-SteveBrush) [inline libs/components/lookup/testing/src/search/search-harness.ts]: This is called several times in the harness. One option is to create an "assert" method that throws an error with additional context to the user:
```
async #assertIsExpanded(errorMessage: string): Promise<void> {
  if (await this.isCollapsed()) {
    throw new Error(`${errorMessage} Search is collapsed.`);
  }
}

public async clear(): Promise<void> {
  await this.#assertIsExpanded('Failed to clear the input.');

  // ...
}
```

File: libs/components/lookup/testing/src/search/search-harness.ts
Comment (Blackbaud-SteveBrush) [inline libs/components/lookup/testing/src/search/search-harness.ts]: Let's name this `clickOpenSearchButton` to match the pattern used for the `clickDismissSearchButton` method, above.

File: apps/code-examples/src/app/code-examples/lookup/search/basic/demo.component.spec.ts
Comment (Blackbaud-SteveBrush) [inline apps/code-examples/src/app/code-examples/lookup/search/basic/demo.component.spec.ts]: 🤔 We export the search component without a lambda because of an issue with list builder; however, the consumer should never use it directly. Is there a way to refactor the unit test to not require it?

Comment (Blackbaud-SandhyaRajasabeson) [reply apps/code-examples/src/app/code-examples/lookup/search/basic/demo.component.spec.ts]: i can remove that import if i remove the unit test showing consumers how to shrink their window size in unit tests which is needed for some of the harness functions, do we think we're ok not showing them that?

Comment (Blackbaud-SteveBrush) [reply apps/code-examples/src/app/code-examples/lookup/search/basic/demo.component.spec.ts]: If I remove the `overrideComponent` call the tests still pass... Is it needed? 🤷‍♂️

File: apps/code-examples/src/app/code-examples/lookup/search/basic/demo.component.spec.ts
Comment (Blackbaud-SteveBrush) [inline apps/code-examples/src/app/code-examples/lookup/search/basic/demo.component.spec.ts]: This type is also marked as "internal". I'm going to work on creating a testing controller for the media query stuff that we can incorporate into our code examples. In the meantime, do you think you can only add tests for things that don't require the media query service?

Comment (Blackbaud-SteveBrush) [reply apps/code-examples/src/app/code-examples/lookup/search/basic/demo.component.spec.ts]: Edit: I went ahead and created the controller this afternoon. Let's wait until it gets merged to proceed on this PR. Take a look (specifically at the search specs): https://github.com/blackbaud/skyux/pull/2802

**Review by Blackbaud-SteveBrush — APPROVED** [review]

File: apps/code-examples/src/app/code-examples/lookup/search/basic/demo.component.spec.ts
Comment (Blackbaud-SandhyaRajasabeson) [reply apps/code-examples/src/app/code-examples/lookup/search/basic/demo.component.spec.ts]: from dms: will fix merge conflicts after pr is merged

**Pull request merged**
